core/src/main/java/com/radixdlt/consensus/sync/BFTSync.java

```diff
@@ -240,8 +251,50 @@ public SyncResult syncToQC(HighQC highQC, @Nullable NodeId author) {
     return switch (vertexStore.insertQc(qc)) {
       case VertexStore.InsertQcResult.Inserted inserted -> {
         // QC was inserted, try TC too (as it can be higher), and then process a new highQC
-        highQC.highestTC().map(vertexStore::insertTimeoutCertificate);
-        this.pacemakerReducer.processQC(vertexStore.highQC());
+        final var tcAlsoInserted =
+            highQC.highestTC().map(vertexStore::insertTimeoutCertificate).orElse(false);
+
+        this.pacemakerReducer
+            .processQC(vertexStore.highQC())
+            .ifPresent(
+                roundUpdate -> {
+                  CertificateType certificateType;
+                  if (tcAlsoInserted) {
+                    // TC was inserted so its round must be higher than that of a QC, round change
+                    // cert type is then a TC
+                    certificateType = CertificateType.TC;
+                  } else {
+                    // QC was inserted, need to determine whether its corresponding vertex was a
+                    // timeout vertex or not
+                    final var isTimeoutVertex =
+                        vertexStore
+                            .getExecutedVertex(qc.getProposedHeader().getVertexId())
+                            .orElseThrow(
+                                () -> new RuntimeException("Vertex is missing for inserted QC"))
+                            .vertex()
+                            .isTimeout();
+                    certificateType =
+                        isTimeoutVertex
+                            ? CertificateType.QC_ON_TIMEOUT_VERTEX
+                            : CertificateType.QC_ON_REGULAR_VERTEX;
+                  }
+
+                  metrics
+                      .bft()
+                      .roundChanges()
+                      .label(
+                          new Metrics.RoundChange(
+                              roundUpdate.getLeader().getKey().toHex(),
+                              roundUpdate
+                                  .getLeader()
+                                  .getValidatorAddress()
+                                  .map(addressing::encodeValidatorAddress)
+                                  .orElse("none"),
+                              highQcSource,
+                              certificateType))
+                      .inc();
+                });
+
```

Comment:
please extract this logic to some method (which makes it clear that its only purpose is to bump a metric)